test cases/common/6 linkshared/meson.build

```diff
@@ -1,4 +1,14 @@
-project('shared library linking test', 'c', 'cpp')
+project('shared library linking test', 'c')
+
+if add_languages('cpp', required: false)
+  cpp = meson.get_compiler('cpp')
+else
+  error('MESON_SKIP_TEST No C++ compiler available.')
+endif
+
+if host_machine.system() == 'windows' and fcid == 'pgi'
+  error('MESON_SKIP_TEST: PGI on Windows does not have a C++ compiler')
+endif
```

Comment:
ditto for all of these where C++ isn't a hard requirement for the test (the "C++ and C" and "C++ and FORTRAN" tests obviously do need C++ to give any useful information).

Reply:
Sorry I got a little careless with the copy and paste

Reply:
no worries, stuff like this is easy to miss and it's why we do code review :)